chain/ethevents/handlers.go

```diff
@@ -139,6 +139,55 @@ func HandleVochainOracle(ctx context.Context, event *ethtypes.Log, e *EthereumEv
 		}
 		log.Debugf("broadcasting tx: %s", log.FormatProto(setProcessTx))
 
+		res, err := e.VochainApp.SendTX(tx)
+		if err != nil || res == nil {
+			log.Warnf("cannot broadcast tx: (%s)", err)
+			return fmt.Errorf("cannot broadcast tx: %w, res: %+v", err, res)
+		}
+		log.Infof("oracle transaction sent, hash: %s", res.Hash)
+
+	case ethereumEventList["processesCensusUpdated"]:
+		tctx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
+		setProcessTx, err := processCensusUpdatedMeta(tctx, &e.ContractsABI[0], event.Data, e.VotingHandle)
+		if err != nil {
+			return err
+		}
+		log.Infof("found process %x census update on ethereum", setProcessTx.ProcessId)
+		p, err := e.VochainApp.State.Process(setProcessTx.ProcessId, true)
+		if err != nil {
+			return err
+		}
+
+		// check dynamic census enabled
+		if !p.Mode.DynamicCensus {
+			return fmt.Errorf("process needs dynamic census in order to update its census")
+		}
+		// check status
+		if (p.Status != models.ProcessStatus_READY) && (p.Status != models.ProcessStatus_PAUSED) {
+			return fmt.Errorf("process status %s does not accept census updates", p.Status.String())
+		}
+		// check census origin
+		if (p.CensusOrigin != models.CensusOrigin_OFF_CHAIN_CA) && (p.CensusOrigin != models.CensusOrigin_OFF_CHAIN_TREE) && (p.CensusOrigin != models.CensusOrigin_OFF_CHAIN_TREE_WEIGHTED) {
```

Comment:
Please do not add this kind of static checks here... You can extend the types.CensusOrigins struct and add a new flag for `allowCensusUpdate`

Reply:
yep, code will be less verbose and cleaner, changed!